core_lib/src/structure/object.cpp

```diff
@@ -164,11 +164,8 @@ void Object::createWorkingDir()
         QFileInfo fileInfo(mFilePath);
         strFolderName = fileInfo.completeBaseName();
     }
-    const QString strWorkingDir = QDir::tempPath()
-        + "/Pencil2D/"
-        + strFolderName
-        + PFF_TMP_DECOMPRESS_EXT
-        + "/";
+    const QString strWorkingDir = 
```

Comment:
The default folders are not getting deleted. This is because the `Object` destructor only calls `deleteWorkingDir` if `filePath()` is empty. Instead I think we should always call `deleteWorkingDir`, and just add a check in that function to make sure that `mWorkingDirPath` is not empty.

On a slightly unrelated note, is there any reason why we don't just call `init` in the `Object` constructor? That way we can be sure that it is called at least once.

Reply:
The behaviours were introduced in a PR #591, I don't really remember why it's working in this way. So the working directory will be deleted anyway now.

For my understanding, Object::init() is for a new Object. In Pencil2D it's the starting default project and the project when you click from the menu File->New. It's no harm to call init() on an Object from a pclx file but all data will be overwritten immediately.